### PR TITLE
send event to save state after 'Add Project Folder' or 'Remove Projec…

### PR DESCRIPTION
**Pull request by hongyuanw**
Fixes https://github.com/atom/atom/issues/12495

When we add or remove project folder,  the window's current project path will be changed, then "initialPaths" will be reloaded, it is the best time to save state

/cc @maxbrunsfeld @lee-dohm

### CODE REVIEW

File: src/application-delegate.coffee [hunk 1]
```diff
@@ -126,6 +126,8 @@ class ApplicationDelegate
     loadSettings = getWindowLoadSettings()
     loadSettings['initialPaths'] = paths
     setWindowLoadSettings(loadSettings)
+    ipcRenderer.send("did-save-state")
```
Comment (maxbrunsfeld) [inline src/application-delegate.coffee]: I think we should rename the event from `did-save-state` to `did-change-paths`. The current name `did-save-state` makes it sound like Atom already *did* save the state.

**Comment by maxbrunsfeld** [comment]
This looks great to me; I left one minor comment. I would like to add a test for this behavior. @yuanwhy are you interested in adding a test for this? I think that it should be added to [this test file](https://github.com/atom/atom/blob/master/spec/main-process/atom-application.test.js).

**Comment by yuanwhy** [comment]
@maxbrunsfeld I renamed the event name. I will appreciate it if you could add a test for this behavior, because I am not good at the test code

File: src/application-delegate.coffee [hunk 2]
```diff
@@ -126,6 +126,8 @@ class ApplicationDelegate
     loadSettings = getWindowLoadSettings()
     loadSettings['initialPaths'] = paths
     setWindowLoadSettings(loadSettings)
+    ipcRenderer.send("did-change-paths")
+
```
Comment (thomasjo) [inline src/application-delegate.coffee]: 🔥 this empty line

**Pull request merged**